app/controllers/submissions_controller.rb

```diff
@@ -124,6 +124,8 @@ def populate_file_manager
       end
       entries = get_all_file_data(revision, grouping, '')
     end
+    entries.append({ only_required_files: assignment.only_required_files,
+                     required_files: assignment.assignment_properties_config['assignment_files_attributes'] })
```

Comment:
`assignment_properties_config` is a heavy-duty method; while it's creative that you found it, a much more efficient approach is to use a query to explicitly extract the `filename`s of the associated `assignment_files` for this `assignment`. (Remember that you only need a list of the filenames, not ids or other attributes.)